packages/web-embed-api/src/index.ts

```diff
@@ -38,22 +39,43 @@ const DEFAULT_CONFIG: IFormsortWebEmbedConfig = {
   origin: DEFAULT_FLOW_ORIGIN,
 };
 
-interface RedirectParams {
+export interface BaseEventProps {
+  answers: IFlowAnswers | undefined;
+}
+export interface RedirectParams extends BaseEventProps {
   url: string;
 }
 
-export interface IEventMap {
-  flowloaded?: () => void;
-  flowclosed?: () => void;
-  flowfinalized?: () => void;
-  redirect?: ({
-    url,
-  }: RedirectParams) =>
-    | {
-        cancel?: boolean;
-        customUrl?: string;
-      }
-    | undefined;
+const supportedAnalyticsEvents = [
+  AnalyticsEventType.FlowLoaded,
+  AnalyticsEventType.FlowClosed,
+  AnalyticsEventType.FlowFinalized,
+  AnalyticsEventType.StepLoaded,
+  AnalyticsEventType.StepCompleted,
+] as const;
+
+type SupportedAnalyticsEvent = typeof supportedAnalyticsEvents[number];
+
+const isSupportedEventType = (
+  eventType: AnalyticsEventType
+): eventType is SupportedAnalyticsEvent =>
+  supportedAnalyticsEvents.includes(eventType as SupportedAnalyticsEvent);
+
+export interface IAnalyticsEventMap {
+  FlowLoaded?: (props: BaseEventProps) => void;
+  FlowClosed?: (props: BaseEventProps) => void;
+  FlowFinalized?: (props: BaseEventProps) => void;
+  StepLoaded?: (props: BaseEventProps) => void;
+  StepCompleted?: (props: BaseEventProps) => void;
+}
+
+export interface IEventMap extends IAnalyticsEventMap {
+  redirect?: (
+    props: RedirectParams
+  ) => {
+    cancel?: boolean;
+    customUrl?: string;
+  } | void;
```

Comment:
Changes from union with `undefined` to union with `void` to allow the user of the callback to _not_ return a value. For `undefined`, Typescript requires that the caller explicitly return the value `undefined` from the callback.